internal/driver/checkstatuses.go

```diff
@@ -25,8 +25,23 @@ func (d *Driver) checkStatuses() {
 
 		status := d.testConnectionMethods(device)
 
-		if err := d.updateDeviceStatus(device.Name, status); err != nil {
+		if statusChanged, err := d.updateDeviceStatus(device.Name, status); err != nil {
 			d.lc.Warnf("Could not update device status for device %s: %s", device.Name, err.Error())
+		} else if statusChanged && status == UpWithAuth {
+			d.lc.Infof("Device %s is now %s, refreshing the device information.", device.Name, UpWithAuth)
+			go func(device models.Device) {
+				err2 := d.refreshDeviceInformation(device)
```

Comment:
Could this keep the name err? Or should the name be more descriptive if it does need to be named differently?